tools/scripts/print_failures.py

```diff
@@ -9,15 +9,15 @@
 if len(sys.argv) != 2:
     sys.exit(f"Usage: python {sys.argv[0]} TEST_LOG_FILENAME")
 
-test_log = json.load(open(sys.argv[1]))
-
-for result in test_log["tests"]:
-    if not result["code"] in ["PASS", "UNSUPPORTED", "XFAIL"]:
-        print("code: {}".format(result["code"]))
-        # Ignore result["elapsed"].
-        print("name: {}".format(result["name"]))
-        # The JSON contains embedded CRLFs (which aren't affected by opening the file in text mode).
-        # If we don't replace these CRLFs with LFs here, this script will appear to be okay in the console,
-        # but redirecting it to a file will result in ugly double newlines.
-        print("output: {}".format(result["output"].replace("\r\n", "\n")))
-        print("==================================================")
+with open(sys.argv[1]) as file:
+    test_log = json.load(file)

```

Comment:
I think there's a shortcut for this, but I forget what it is :(

Reply:
If you remember later, I can go back and simplify this! :smile_cat: